packages/backend/src/tokens/keys.ts

```diff
@@ -0,0 +1,207 @@
+import { API_URL, API_VERSION } from '../constants';
+// DO NOT CHANGE: Runtime needs to be imported as a default export so that we can stub its dependencies with Sinon.js
+// For more information refer to https://sinonjs.org/how-to/stub-dependency/
+import runtime from '../runtime';
+import { callWithRetry } from '../util/callWithRetry';
+import { joinPaths } from '../util/path';
+import { TokenVerificationError, TokenVerificationErrorAction, TokenVerificationErrorReason } from './errors';
+
+type JsonWebKeyWithKid = JsonWebKey & { kid: string };
+
+type JsonWebKeyCache = Record<string, JsonWebKeyWithKid>;
+
+let cache: JsonWebKeyCache = {};
```

Comment:
Since this is running in a server environment, would it be better if we make this a `WeakMap` instead?

Reply:
Actually scratch that, this is the JWK cache and keys have TTL so its not going to get big anyway. (I still think that a `Map` is generally better suited for a cache mainly because of `cache.clear()` but that's not important)

Reply:
You are right. A map is better. I should have used it in the first place. I will do a follow up PR.